routers/api/v1/repo/wiki.go

```diff
@@ -0,0 +1,501 @@
+		return
+	}
+
+	ctx.Status(http.StatusNoContent)
+}
+
+// WikiPages get wiki pages list
+func WikiPages(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/wiki/pages repository repoGetWikiPages
+	// ---
+	// summary: Get all wiki pages
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// - name: limit
+	//   in: query
+	//   description: page size of results
+	//   type: integer
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/WikiPageList"
+
+	wikiRepo, commit, err := findWikiRepoCommit(ctx)
+	if err != nil {
+		if wikiRepo != nil {
+			wikiRepo.Close()
+		}
+		return
+	}
+	defer func() {
+		if wikiRepo != nil {
+			wikiRepo.Close()
+		}
+	}()
+
+	page := ctx.FormInt("page")
+	if page <= 1 {
+		page = 1
+	}
+
+	limit := ctx.FormInt("limit")
+	if limit <= 1 {
+		limit = 20
+	}
+
+	skip := (page - 1) * limit
+	max := page * limit
+
+	entries, err := commit.ListEntries()
+	if err != nil {
+		ctx.ServerError("ListEntries", err)
+		return
+	}
+	pages := make([]api.PageMeta, 0, len(entries))
+	for i, entry := range entries {
+		if i < skip || i >= max || !entry.IsRegular() {
+			continue
+		}
+		c, err := wikiRepo.GetCommitByPath(entry.Name())
+		if err != nil {
+			ctx.Error(http.StatusInternalServerError, "GetCommit", err)
+			return
+		}
+		wikiName, err := wiki_service.FilenameToName(entry.Name())
+		if err != nil {
+			if models.IsErrWikiInvalidFileName(err) {
+				continue
+			}
+			ctx.Error(http.StatusInternalServerError, "WikiFilenameToName", err)
+			return
+		}
+		pages = append(pages, api.PageMeta{
+			Name:    wikiName,
+			SubURL:  wiki_service.NameToSubURL(wikiName),
+			Updated: c.Author.When.Format(time.RFC3339),
+		})
+	}
+
+	ctx.JSON(http.StatusOK, pages)
+}
+
+// Wiki get single wiki page
+func Wiki(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/wiki/page/{pageName} repository repoGetWikiPage
+	// ---
+	// summary: Get a wiki page
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: pageName
+	//   in: path
+	//   description: name of the page
+	//   type: string
+	//   required: true
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/WikiPage"
+
+	wikiRepo, commit, err := findWikiRepoCommit(ctx)
+	if err != nil {
+		if wikiRepo != nil {
+			wikiRepo.Close()
+		}
+		if !git.IsErrNotExist(err) {
+			ctx.Error(http.StatusInternalServerError, "GetBranchCommit", err)
+		}
```

Comment:
```suggestion
		if git.IsErrNotExist(err) {
			ctx.Error(http.StatusInternalServerError, "GetBranchCommit", err)
		} else {
			ctx.Error(http.StatusInternalServerError, "findWikiRepoCommit", err)
		}
```
Or am I misunderstanding something here?

Reply:
see https://github.com/go-gitea/gitea/pull/17278/commits/d3ef69c5bd1db3b834eca037765aa450f0800975